### PR TITLE
Add CreateDomain acceptance tests

### PR DESCRIPTION
**Pull request by l4l**
### Description of the Change

Add CreateDomain tests.

<s>Hopefully all works as expected, so there is no disabled tests</s>
**UPD**: empty role passes stateless validation, so it is disabled

### Benefits

Coverage & stability

### Possible Drawbacks 

+ ~100 sec of test execution time :(

### Usage Examples or Tests

`make create_domain_test && test_bin/create_domain_test`

### CODE REVIEW

File: test/integration/acceptance/create_domain_test.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+ public:
+  /**
+   * Creates the transaction with the user creation commands
+   * @param perms are the permissions of the user
+   * @return built tx
+   */
+  auto makeUserWithPerms(const std::vector<std::string> &perms = {
+                             iroha::model::can_create_domain}) {
+    auto new_perms = perms;
+    new_perms.push_back(iroha::model::can_set_quorum);
```
Comment (Alexey-N-Chernyshov) [inline test/integration/acceptance/create_domain_test.cpp]: please, remove if it is doesn't needed here

File: test/integration/acceptance/create_domain_test.cpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+   */
+  auto baseTx() {
+    return TestUnsignedTransactionBuilder()
+        .txCounter(1)
+        .creatorAccountId(kUserId)
+        .createdTime(iroha::time::now());
+  }
+
+  /**
+   * Completes pre-build transaction
```
Comment (Alexey-N-Chernyshov) [inline test/integration/acceptance/create_domain_test.cpp]: built?

File: test/integration/acceptance/create_domain_test.cpp [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+      .sendTx(completeTx(baseTx().createDomain(kNewDomain, kRole)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with inexistent domain
```
Comment (Alexey-N-Chernyshov) [inline test/integration/acceptance/create_domain_test.cpp]: inexistent role

Comment (Alexey-N-Chernyshov) [reply test/integration/acceptance/create_domain_test.cpp]: I don't know may be you should add test for non existing domain?

File: test/integration/acceptance/create_domain_test.cpp [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with maximum available length
+ * @then there is no tx in proposal
```
Comment (Alexey-N-Chernyshov) [inline test/integration/acceptance/create_domain_test.cpp]: there is tx in proposal

File: test/integration/acceptance/create_domain_test.cpp [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        .createdTime(iroha::time::now());
+  }
+
+  /**
+   * Completes pre-build transaction
+   * @param builder is a pre-built tx
+   * @return built tx
+   */
+  template <typename TestTransactionBuilder>
+  auto completeTx(TestTransactionBuilder builder) {
```
Comment (nickaleks) [inline test/integration/acceptance/create_domain_test.cpp]: Why pass by value?

**Comment by nickaleks** [comment]
Can you explain why tests take such a long time? And what can be done to solve that issue

File: test/integration/acceptance/create_domain_test.cpp (hunk 3 above)
Comment (igor-egorov) [reply test/integration/acceptance/create_domain_test.cpp]: Please fix `@when` part of docstring and column for C241 in [Confluence](https://soramitsu.atlassian.net/wiki/spaces/IS/pages/299958273/Acceptance+Tests).

File: test/integration/acceptance/create_domain_test.cpp (hunk 4 above)
Comment (igor-egorov) [reply test/integration/acceptance/create_domain_test.cpp]: Please fix `@then` part of docstring and corresponding column for C243.1 in [Confluence](https://soramitsu.atlassian.net/wiki/spaces/IS/pages/299958273/Acceptance+Tests).

File: test/integration/acceptance/create_domain_test.cpp [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  ASSERT_ANY_THROW(itf.skipProposal());
+}
+
+/**
+ * @given some user with can_create_domain permission
+ * @when execute tx with CreateDomain command with empty role name
+ * @then the tx hasn't passed stateless validation
+ *       (aka skipProposal throws)
+ */
+TEST_F(CreateDomain, DISABLED_EmptyRole) {
```
Comment (igor-egorov) [inline test/integration/acceptance/create_domain_test.cpp]: Could you please file a bug in Jira for the issue?

File: test/integration/acceptance/create_domain_test.cpp (hunk 5 above)
Comment (l4l) [reply test/integration/acceptance/create_domain_test.cpp]: It is a template param

**Comment by l4l** [comment]
@nickaleks here's the issue
https://github.com/hyperledger/iroha/blob/e8d765f9f20dc06be6c7175d27433c3539cabd49/test/framework/integration_framework/integration_test_framework.hpp#L121-L126

So basically all of the missing proposal/block checks are required to wait such a time

File: test/integration/acceptance/create_domain_test.cpp (hunk 5 above)
Comment (nickaleks) [reply test/integration/acceptance/create_domain_test.cpp]: You can pass template param reference

File: test/integration/acceptance/create_domain_test.cpp [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
```
Comment (lebdron) [inline test/integration/acceptance/create_domain_test.cpp]: Please rename EmptyRole to EmptyRoleName, also in other descriptions. EmptyRole can also relate to role having no permissions.

File: test/integration/acceptance/create_domain_test.cpp (hunk 5 above)
Comment (l4l) [reply test/integration/acceptance/create_domain_test.cpp]: Yes, but it won't change much

Comment (nickaleks) [reply test/integration/acceptance/create_domain_test.cpp]: Okay, it is fine, I was interested if this was a special feature I was not aware of :)

**Pull request merged**